src/PokemonApp.js

```diff
@@ -1,29 +1,41 @@
-import {ThemeProvider} from "@material-ui/core";
-import {Switch, Route} from 'react-router-dom';
-import {ErrorBoundary} from "./error/ErrorBoundary";
-import {theme} from "./theme";
+import { ThemeProvider } from "@material-ui/core/";
+import ScopedCssBaseline from "@material-ui/core/ScopedCssBaseline";
+import { Redirect, Route, Switch } from "react-router-dom";
+import { ErrorBoundary } from "./error/ErrorBoundary";
+import { theme } from "./theme";
 
 import Header from "./components/Header/Header.component";
 import Footer from "./components/Footer/Footer.component";
 import PokemonTablePage from "./pages/PokemonTablePage/PokemonTablePage";
 import Homepage from "./pages/Homepage/Homepage";
-import DownloadPokemonForm from "./components/DownloadPokemonForm/DownloadPokemonForm";
+import SignInPage from "./pages/SignInPage/SignInPage";
 import React from "react";
+import { useSelector } from "react-redux";
+import SignUpPage from "./pages/SignUpPage/SignUpPage";
 
-function PokemonApp() {
-    return (
-        <ErrorBoundary>
-            <ThemeProvider theme={theme}>
-                <Header/>
-                <Switch>
-                    <Route exact path='/' component={Homepage}/>
-                    <Route exact path='/pokemon' component={PokemonTablePage}/>
-                </Switch>
-                <DownloadPokemonForm/>
-                <Footer/>
-            </ThemeProvider>
-        </ErrorBoundary>
-    );
-}
-
-export default PokemonApp;
\ No newline at end of file
+export const PokemonApp = () => {
+  const isAuthenticated = useSelector((state) => state.login.isAuthenticated);
+  return (
+    <ScopedCssBaseline>
+      <ErrorBoundary>
+        <ThemeProvider theme={theme}>
+          <Header />
+          {isAuthenticated ? (
+            <Switch>
+              <Route exact path="/" component={Homepage} />
+              <Route exact path="/pokemon" component={PokemonTablePage} />
+              <Redirect to="/login/signin" />
+            </Switch>
+          ) : (
+            <Switch>
```

Comment:
Probably it's better to use PrivateRoute instead of two different Switch

Reply:
Done. Added Private and Public routes. I agree with you. It looks cleaner.